pkg/local_object_storage/shard/gc.go

```diff
@@ -284,17 +298,25 @@ func (s *Shard) collectExpiredTombstones(ctx context.Context, e Event) {
 	for {
 		log.Debug("iterating tombstones")
 
-		if s.GetMode().NoMetabase() {
+		s.m.RLock()
```

Comment:
Same here.

Reply:
`expiredTombstonesCallback` could call the same shard and deadlock once again